enterprise/frontend/src/embedding-sdk/components/public/dashboard/SdkDashboard.tsx

```diff
@@ -151,6 +157,12 @@ const SdkDashboardInner = ({
     dashboardId,
   });
 
+  const [renderModeState, setRenderMode] =
+    useState<Extract<RenderMode, "dashboard" | "queryBuilder">>("dashboard");
```

Comment:
nit: `"dashboard" | "queryBuilder"` would suffice as you would get a type error in `const finalRenderMode: RenderMode` anyway if the `renderModeState` type changed.

Reply:
Good point, that clean things up. Fixed in a15838f7af333a7824fca8d0f443ba64a3509248